### PR TITLE
Fix NavBar mess

### PR DESCRIPTION
**Pull request by mjlumetta**
https://github.com/department-of-veterans-affairs/vets-contrib/issues/2893

### CODE REVIEW

File: config/webpack.config.prod.js [hunk 1]
```diff
@@ -458,7 +458,8 @@ module.exports = (envName) => {
     },
     performance: {
       hints: 'error',
-      maxAssetSize: 300000,
```
Comment (jasalisbury) [inline config/webpack.config.prod.js]: Why did these values have to change?

File: src/App.tsx [hunk 2]
```diff
@@ -15,10 +15,10 @@ import { history } from './store';
...
 import './App.scss';
 
-let currentPath = history.location.pathname;
-history.listen(location => {
-  currentPath = location.pathname;
-});
+// let currentPath = history.location.pathname;
+// history.listen(location => {
+//   currentPath = location.pathname;
+// });
```
Comment (jasalisbury) [inline src/App.tsx]: any reason to comment this snippet instead of removing it?

File: src/components/NavBar.scss [hunk 3]
```diff
@@ -1,217 +1,176 @@
 @import '../styles/variables';
 
-.header-content {
-  max-width: none;
+.va-api-nav {
+  background: $color-white;
+  color: #323A45;
+  border-top: 1px solid #DBDBDD;
```
Comment (jasalisbury) [inline src/components/NavBar.scss]: Should we be using a color variable here?

File: src/components/NavBar.test.tsx [hunk 4]
```diff
@@ -6,17 +6,17 @@ import NavBar from './NavBar';
...
 describe('NavBar', () => {
   it('should render the navbar', () => {
-    const documentation = shallow(<NavBar hideLinks={false} />).contains('Documentation');
+    const documentation = shallow(<NavBar />).contains('Documentation');
     expect(documentation).toBeTruthy();
   });
 
   it('should use "is-hidden" when hidelinks is passed', () => {
-    const result = shallow(<NavBar hideLinks={true} />);
+    const result = shallow(<NavBar />);
```
Comment (kalilsn) [inline src/components/NavBar.test.tsx]: I think this test should just be removed entirely since the `hidelinks` functionality was deleted.

File: src/components/Header.scss [hunk 5]
```diff
@@ -0,0 +1,65 @@
+/*
+  This file is here to prep for separating NavBar into two components, NavBar and Header. This file is for 
+  Header, which would currently be the top-level component. It is actually called NavBar, but that naming is
+  confusing, given that the top levels tags and CSS classes use "header" naming terminology. Anyway, in the 
+  future, there will be a component Header to accompany these styles; currently they are used in NavBar.
```
Comment (kalilsn) [inline src/components/Header.scss]: 👏

File: src/containers/ApplyForm.tsx [hunk 6]
```diff
@@ -279,8 +281,8 @@ class ApplyForm extends React.Component<IApplyProps> {
   }
 
   private anyOAuthApisSelected() {
-    const apiUrlFragments = this.selectedApis().flatMap((formField) => formFieldsToFragments[formField]);
-    return includesOauthAPI(apiUrlFragments);
+    const apiIdsByField = this.selectedApis().map((formField) => formFieldsToFragments[formField]);
+    return includesOauthAPI(flatten(apiIdsByField));
```
Comment (kalilsn) [inline src/containers/ApplyForm.tsx]: Is this related to the formation navbar stuff? If not can we put it in a different PR?

File: config/webpack.config.prod.js (hunk 1 above)
Comment (kalilsn) [reply config/webpack.config.prod.js]: 😢

Comment (kalilsn) [reply config/webpack.config.prod.js]: ^ that's assuming this change is because the assets have increased in size

Comment (mjlumetta) [reply config/webpack.config.prod.js]: yeah it isn't specific to this PR it's a result of the USWDS -> VADS change. it may be fixable later?

File: src/App.tsx (hunk 2 above)
Comment (mjlumetta) [reply src/App.tsx]: that was a mistake haha, thanks for catching

File: src/components/NavBar.scss (hunk 3 above)
Comment (mjlumetta) [reply src/components/NavBar.scss]: yes we should, but i don't think it belongs in this PR. i think we should catch all those in one PR

File: src/components/NavBar.test.tsx (hunk 4 above)
Comment (mjlumetta) [reply src/components/NavBar.test.tsx]: yeah, you're right - i hadn't really come back to these after making it compile

File: src/containers/ApplyForm.tsx (hunk 6 above)
Comment (mjlumetta) [reply src/containers/ApplyForm.tsx]: yeah i'll open up a new PR

File: package.json [hunk 7]
```diff
@@ -50,6 +50,7 @@
     "jest-puppeteer": "^4.1.1",
     "js-yaml-loader": "^1.2.2",
     "lodash.debounce": "^4.0.8",
+    "lodash.flatten": "^4.4.0",
```
Comment (kalilsn) [inline package.json]: I think this should to go too (ditto the accompanying change in package.lock)

File: src/App.tsx [hunk 8]
```diff
@@ -9,18 +9,12 @@ import { getEnvFlags } from './apiDefs/env';
...
 
-import './App.scss';
-
-let currentPath = history.location.pathname;
-history.listen(location => {
-  currentPath = location.pathname;
-});
-
 import 'highlight.js/styles/github.css';
+import './App.scss';
```
Comment (kalilsn) [inline src/App.tsx]: Moving this after the highlight.js styles seems like a good choice in general, but was there a bug that this fixed?

File: package.json (hunk 7 above)
Comment (kalilsn) [reply package.json]: Oops! Just realized the actual flatmap change is still in here too :)

File: src/App.tsx (hunk 8 above)
Comment (mjlumetta) [reply src/App.tsx]: nope! this was more remove a blank line -> linter yells @ u to alphabetize things. but because, as you say, it's already a good idea to move it and there shouldn't be collisions, why not

File: package.json (hunk 7 above)
Comment (mjlumetta) [reply package.json]: yup definitely will merge in `master` and resolve those conflicts

File: src/components/Header.scss [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+*/
```
Comment (kalilsn) [inline src/components/Header.scss]: Looks like this comment can actually be removed now
